components/gfx/platform/windows/font.rs

```diff
@@ -40,11 +42,211 @@ impl FontTableMethods for FontTable {
     }
 }
 
+fn make_tag(tag_bytes: &[u8]) -> FontTableTag {
+    assert!(tag_bytes.len() == 4);
+    unsafe { *(tag_bytes.as_ptr() as *const FontTableTag) }
+}
+
+macro_rules! try_lossy(($result:expr) => (try!($result.map_err(|e| (())))));
+
+// Given a set of records, figure out the string indices for the family and face
+// names.  We want name_id 1 and 2, and we need to use platform_id == 1 and
+// language_id == 0 to avoid limitations in the truetype crate.  We *could* just
+// do our own parsing here, and use the offset/length data and pull the values out
+// ourselves.
+fn get_family_face_indices(records: &[truetype::naming_table::Record]) -> Option<(usize, usize)> {
+    let mut family_name_index = None;
+    let mut face_name_index = None;
+
+    for i in 0..records.len() {
+        // the truetype crate can only decode mac platform format names
+        if records[i].platform_id != 1 {
+            continue;
+        }
+
+        if records[i].language_id != 0 {
+            continue;
+        }
+
+        if records[i].name_id == 1 {
+            family_name_index = Some(i);
+        } else if records[i].name_id == 2 {
+            face_name_index = Some(i);
+        }
+    }
+
+    if family_name_index.is_some() && face_name_index.is_some() {
+        Some((family_name_index.unwrap(), face_name_index.unwrap()))
+    } else {
+        None
+    }
+}
+
+// We need the font (DWriteFont) in order to be able to query things like
+// the family name, face name, weight, etc.  On Windows 10, the
+// DWriteFontFace3 interface provides this on the FontFace, but that's only
+// available on Win10+.
+//
+// Instead, we do the parsing work using the truetype crate for raw fonts.
+// We're just extracting basic info, so this is sufficient for now.
+
+#[derive(Debug)]
+struct FontInfo {
+    family_name: String,
+    face_name: String,
+    weight: font_weight::T,
+    stretch: font_stretch::T,
+    style: FontStyle,
+}
+
+impl FontInfo {
+    fn new_from_face(face: &FontFace) -> Result<FontInfo, ()> {
+        use std::cmp::{min, max};
+        use std::io::Cursor;
+        use truetype::Value;
+
+        let mut info = FontInfo {
+            family_name: "".to_owned(),
+            face_name: "".to_owned(),
+            weight: font_weight::T::Weight400,
+            stretch: font_stretch::T::normal,
+            style: FontStyle::Normal,
+        };
+
+        if let Some(mut name_table_bytes) = face.get_font_table(make_tag(b"name")) {
```

Comment:
nit: Maybe this would be a bit cleaner if you do something like:

```rust
let mut name_table_bytes = match face.get_font_table(make_tag(b"name")) {
    Some(table) => table,
    None => return Err(()),
};

// Use name_table_bytes here.
```

This applies also to other places below.